Update libsecp256k1 subtree to current master

sipa opened the pull request:
The motivation for this bump is getting rid of a cast in `CKey::SignSchnorr`; the `aux_rand` argument isn't modified by the `secp256k1_schnorrsig_sign` function, but was marked as non-`const` anyway. This is fixed now (bitcoin-core/secp256k1#966), and the cast is removed in this PR.

There are a few other relevant changes:
* (bitcoin-core/secp256k1#956): replaces a runtime-computed table with a precomputed one; this adds arouns 1 MiB to the binary size, but is a step towards significantly simplifying the API. If 1 MiB is too much, it can be reduced by 2 or 4 (or more) for a slight verification performance reduction.
* (bitcoin-core/secp256k1#983): removes (test/bench only) OpenSSL support entirely, removing the need to pass `--disable-openssl-tests` (see #23314).
* (bitcoin-core/secp256k1#810): mild performance increase for 64-bit non-x86 platforms.
* (bitcoin-core/secp256k1#1002): Make aux_rnd32==NULL behave identical to 0x0000..00 (which impacts BIP341/BIP342 signing in Bitcoin Core, making it more strictly BIP340 compliant, though not in a manner that affects security).

sipa committed:
Remove --disable-openssl-tests for libsecp256k1 configure

sipa committed:
Remove unnecessary cast in CKey::SignSchnorr

sipa commented:
@MarcoFalke Yeah, but I'd like to not wait for that.

sipa commented:
Updated to also include https://github.com/bitcoin-core/secp256k1/pull/1002

fanquake commented:
Guix build:
```bash
bash-5.1# find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
23f5a1226d8473d4a6f17a41276fd023d281b7d6df8d566241d1db46adcce7ee  guix-build-63831b1088fa/output/aarch64-linux-gnu/SHA256SUMS.part
41c1f9c899d829532bbffa055ae6cf12564f64fa037eb80f82c85140049a3ab0  guix-build-63831b1088fa/output/aarch64-linux-gnu/bitcoin-63831b1088fa-aarch64-linux-gnu-debug.tar.gz
c00b0ae5de097e1692e15388522f52c440bcc7103f32151c1fd2a95fa0b4b59f  guix-build-63831b1088fa/output/aarch64-linux-gnu/bitcoin-63831b1088fa-aarch64-linux-gnu.tar.gz
064492c88c98fdbb2c9f8c7bc327e7582dff2e40b7f71be9a689ac43cc2d5d9d  guix-build-63831b1088fa/output/arm-linux-gnueabihf/SHA256SUMS.part
9a0b09113ea43a9dabe7309deacb7f502fc4d1be6ca1bfe4e0e1a312cc9b5fbf  guix-build-63831b1088fa/output/arm-linux-gnueabihf/bitcoin-63831b1088fa-arm-linux-gnueabihf-debug.tar.gz
4935fafcffe8bd505922e5b0741dbb7509818339fa005321d5197201b33c3c85  guix-build-63831b1088fa/output/arm-linux-gnueabihf/bitcoin-63831b1088fa-arm-linux-gnueabihf.tar.gz
ed1b69903a69558768896d26b247f2c1873c819d78a6ddd25e8442929009de2b  guix-build-63831b1088fa/output/dist-archive/bitcoin-63831b1088fa.tar.gz
b4bc19c7a7d7c35142434d9db5dffc348d7b2d7f02f30a1be8a503fb554b1090  guix-build-63831b1088fa/output/powerpc64-linux-gnu/SHA256SUMS.part
675370507507e7a849e18951c0d869cfefabbeb012fd1d152209d731de598f32  guix-build-63831b1088fa/output/powerpc64-linux-gnu/bitcoin-63831b1088fa-powerpc64-linux-gnu-debug.tar.gz
e9152cbe49f491aa6a24f970eec7b4c780931831d4eb5f271b14a0fa2b410441  guix-build-63831b1088fa/output/powerpc64-linux-gnu/bitcoin-63831b1088fa-powerpc64-linux-gnu.tar.gz
2e57374a107e35fc4a2e85171449c8c6b4fe054b287e77f51c7c7c750b481702  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/SHA256SUMS.part
8fc398a218483b82a329d11d7c5b0f0702596a9b0c94eb237dd82127aea16154  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/bitcoin-63831b1088fa-powerpc64le-linux-gnu-debug.tar.gz
4e046ea95b0dd96248dee713f189de8118c5c84951d81530d6b75501ceb4bf83  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/bitcoin-63831b1088fa-powerpc64le-linux-gnu.tar.gz
bc4adc2698226719114f5327eea4f52586911983c5a835a4155efb15e92df8a5  guix-build-63831b1088fa/output/riscv64-linux-gnu/SHA256SUMS.part
8ddd9d8654a88a3846f53d3f752294ec0da0677871bbb0eea5df63e92a76df69  guix-build-63831b1088fa/output/riscv64-linux-gnu/bitcoin-63831b1088fa-riscv64-linux-gnu-debug.tar.gz
e936d8440bc3f3ae1d11e1291fb6c559688fedaae994a18dc8484bf1f954967d  guix-build-63831b1088fa/output/riscv64-linux-gnu/bitcoin-63831b1088fa-riscv64-linux-gnu.tar.gz
a4bc7ed61d666b89918023c03cbd66d83bf7cf412c5c48b1743f1e240a255e74  guix-build-63831b1088fa/output/x86_64-apple-darwin19/SHA256SUMS.part
0f774f56b18174349ed7487d628ee5ead1cf675cb8d78f0232e6214434594b1b  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx-unsigned.dmg
76d4f6de4e4832dd8d44061e8943f835291f8eb5a1f49f4007d662463199e9f9  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx-unsigned.tar.gz
0433e0f4387f040deb8b633b05adc66b79dc3680c0e60f465adba1ae5ae1eed8  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx64.tar.gz
e4bde26cd33893e1b81c9cc5998ef1d7fbb6b5e8db261daaced5d4ffbe5b8885  guix-build-63831b1088fa/output/x86_64-linux-gnu/SHA256SUMS.part
1595f47c5cc1a94896c1980725e66db828bf3832428c353852f7766863744c7c  guix-build-63831b1088fa/output/x86_64-linux-gnu/bitcoin-63831b1088fa-x86_64-linux-gnu-debug.tar.gz
89f719af33460e77f6b185d017c3735da47bd04b030fb289e0c04e083c427288  guix-build-63831b1088fa/output/x86_64-linux-gnu/bitcoin-63831b1088fa-x86_64-linux-gnu.tar.gz
49ba39976a1f14e71979278691eb6a0079caff7101e4607bfd4ef2fd8789370d  guix-build-63831b1088fa/output/x86_64-w64-mingw32/SHA256SUMS.part
14eef5d925ffbb46023c9c6b021b2bfdf901261beec506bf2431c03a70793907  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win-unsigned.tar.gz
4ef7d01390a372dea976cd88bf7e03b1a39b236b06a36a0100f008ee975c43f6  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64-debug.zip
4b2db6b29570822cd18eae2cbf01a879f526e9cc4a70588ae7dea4804a492be0  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64-setup-unsigned.exe
938215dfc3c6fdff2d7ccd074426d6a2b9c37aa392ffac1ecb54f19bc2221b08  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64.zip
```

hebasto commented:
#### Guix builds
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
23f5a1226d8473d4a6f17a41276fd023d281b7d6df8d566241d1db46adcce7ee  guix-build-63831b1088fa/output/aarch64-linux-gnu/SHA256SUMS.part
41c1f9c899d829532bbffa055ae6cf12564f64fa037eb80f82c85140049a3ab0  guix-build-63831b1088fa/output/aarch64-linux-gnu/bitcoin-63831b1088fa-aarch64-linux-gnu-debug.tar.gz
c00b0ae5de097e1692e15388522f52c440bcc7103f32151c1fd2a95fa0b4b59f  guix-build-63831b1088fa/output/aarch64-linux-gnu/bitcoin-63831b1088fa-aarch64-linux-gnu.tar.gz
064492c88c98fdbb2c9f8c7bc327e7582dff2e40b7f71be9a689ac43cc2d5d9d  guix-build-63831b1088fa/output/arm-linux-gnueabihf/SHA256SUMS.part
9a0b09113ea43a9dabe7309deacb7f502fc4d1be6ca1bfe4e0e1a312cc9b5fbf  guix-build-63831b1088fa/output/arm-linux-gnueabihf/bitcoin-63831b1088fa-arm-linux-gnueabihf-debug.tar.gz
4935fafcffe8bd505922e5b0741dbb7509818339fa005321d5197201b33c3c85  guix-build-63831b1088fa/output/arm-linux-gnueabihf/bitcoin-63831b1088fa-arm-linux-gnueabihf.tar.gz
ed1b69903a69558768896d26b247f2c1873c819d78a6ddd25e8442929009de2b  guix-build-63831b1088fa/output/dist-archive/bitcoin-63831b1088fa.tar.gz
b4bc19c7a7d7c35142434d9db5dffc348d7b2d7f02f30a1be8a503fb554b1090  guix-build-63831b1088fa/output/powerpc64-linux-gnu/SHA256SUMS.part
675370507507e7a849e18951c0d869cfefabbeb012fd1d152209d731de598f32  guix-build-63831b1088fa/output/powerpc64-linux-gnu/bitcoin-63831b1088fa-powerpc64-linux-gnu-debug.tar.gz
e9152cbe49f491aa6a24f970eec7b4c780931831d4eb5f271b14a0fa2b410441  guix-build-63831b1088fa/output/powerpc64-linux-gnu/bitcoin-63831b1088fa-powerpc64-linux-gnu.tar.gz
2e57374a107e35fc4a2e85171449c8c6b4fe054b287e77f51c7c7c750b481702  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/SHA256SUMS.part
8fc398a218483b82a329d11d7c5b0f0702596a9b0c94eb237dd82127aea16154  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/bitcoin-63831b1088fa-powerpc64le-linux-gnu-debug.tar.gz
4e046ea95b0dd96248dee713f189de8118c5c84951d81530d6b75501ceb4bf83  guix-build-63831b1088fa/output/powerpc64le-linux-gnu/bitcoin-63831b1088fa-powerpc64le-linux-gnu.tar.gz
bc4adc2698226719114f5327eea4f52586911983c5a835a4155efb15e92df8a5  guix-build-63831b1088fa/output/riscv64-linux-gnu/SHA256SUMS.part
8ddd9d8654a88a3846f53d3f752294ec0da0677871bbb0eea5df63e92a76df69  guix-build-63831b1088fa/output/riscv64-linux-gnu/bitcoin-63831b1088fa-riscv64-linux-gnu-debug.tar.gz
e936d8440bc3f3ae1d11e1291fb6c559688fedaae994a18dc8484bf1f954967d  guix-build-63831b1088fa/output/riscv64-linux-gnu/bitcoin-63831b1088fa-riscv64-linux-gnu.tar.gz
a4bc7ed61d666b89918023c03cbd66d83bf7cf412c5c48b1743f1e240a255e74  guix-build-63831b1088fa/output/x86_64-apple-darwin19/SHA256SUMS.part
0f774f56b18174349ed7487d628ee5ead1cf675cb8d78f0232e6214434594b1b  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx-unsigned.dmg
76d4f6de4e4832dd8d44061e8943f835291f8eb5a1f49f4007d662463199e9f9  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx-unsigned.tar.gz
0433e0f4387f040deb8b633b05adc66b79dc3680c0e60f465adba1ae5ae1eed8  guix-build-63831b1088fa/output/x86_64-apple-darwin19/bitcoin-63831b1088fa-osx64.tar.gz
e4bde26cd33893e1b81c9cc5998ef1d7fbb6b5e8db261daaced5d4ffbe5b8885  guix-build-63831b1088fa/output/x86_64-linux-gnu/SHA256SUMS.part
1595f47c5cc1a94896c1980725e66db828bf3832428c353852f7766863744c7c  guix-build-63831b1088fa/output/x86_64-linux-gnu/bitcoin-63831b1088fa-x86_64-linux-gnu-debug.tar.gz
89f719af33460e77f6b185d017c3735da47bd04b030fb289e0c04e083c427288  guix-build-63831b1088fa/output/x86_64-linux-gnu/bitcoin-63831b1088fa-x86_64-linux-gnu.tar.gz
49ba39976a1f14e71979278691eb6a0079caff7101e4607bfd4ef2fd8789370d  guix-build-63831b1088fa/output/x86_64-w64-mingw32/SHA256SUMS.part
14eef5d925ffbb46023c9c6b021b2bfdf901261beec506bf2431c03a70793907  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win-unsigned.tar.gz
4ef7d01390a372dea976cd88bf7e03b1a39b236b06a36a0100f008ee975c43f6  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64-debug.zip
4b2db6b29570822cd18eae2cbf01a879f526e9cc4a70588ae7dea4804a492be0  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64-setup-unsigned.exe
938215dfc3c6fdff2d7ccd074426d6a2b9c37aa392ffac1ecb54f19bc2221b08  guix-build-63831b1088fa/output/x86_64-w64-mingw32/bitcoin-63831b1088fa-win64.zip
```

real-or-random commented:
Concept ACK on updating the tree

laanwj commented:
I think this is ready for merge, needs rebase for a conflict in `src/key.cpp` though.

sipa commented:
@laanwj Will address.

sipa commented:
Rebased, and updated to latest master.

Newly included changes:
* bitcoin-core/secp256k1#1019: build: don't append valgrind CPPFLAGS if not installed (macOS)
* bitcoin-core/secp256k1#1020: doc: remove use of <0xa0> "no break space"
* bitcoin-core/secp256k1#1010: doc: Minor fixes in safegcd_implementation.md
* bitcoin-core/secp256k1#1012: Fix typos
* bitcoin-core/secp256k1#1011: ci: Enable -g if we set CFLAGS manually
* bitcoin-core/secp256k1#1009: refactor: Use (int)&(int) in boolean context to avoid compiler warning
* bitcoin-core/secp256k1#1007: doc: Replace apoelstra's GPG key by jonasnick's GPG key
* bitcoin-core/secp256k1#991: Merge all "external" benchmarks into a single bench binary

(all documentation/CI/bench changes)

real-or-random commented:
> Is the plan to do another bump shortly after, after [bitcoin-core/secp256k1#988](https://github.com/bitcoin-core/secp256k1/pull/988), to fix #22854 ?

Now we could consider waiting for it... https://github.com/bitcoin-core/secp256k1/pull/988 is really close to having to ACKs which I think would be good enough to get it merged. @sipa  What do you think?

fanquake commented:
Guix build:
```bash
bash-5.1# find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
cc137780b4627e896a7d4995c713938a94d7dad572c96e0b85fdd7ff368ca137  guix-build-89e24fa6a670/output/aarch64-linux-gnu/SHA256SUMS.part
48b964bd125987f2d2ad5718db564fc35fa0d8466464d78c5b3f3f7e7a0f4b6a  guix-build-89e24fa6a670/output/aarch64-linux-gnu/bitcoin-89e24fa6a670-aarch64-linux-gnu-debug.tar.gz
1ee6fe2021d4e0fb22048c63085f2701c678705def890671e20593adb030b882  guix-build-89e24fa6a670/output/aarch64-linux-gnu/bitcoin-89e24fa6a670-aarch64-linux-gnu.tar.gz
794d048626880d23ca8dafd0d199bc0ebb2d8d1bff6b6b43697c4cdc3c66ce4a  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/SHA256SUMS.part
90d91e3401f2d2182733d35570747fdb4499048f8c78fedf853f253031fbb83c  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/bitcoin-89e24fa6a670-arm-linux-gnueabihf-debug.tar.gz
eb2830ddd5303a6daa77f823dbc19ea7a8f0215aae1750d8b1c831adfa2e149a  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/bitcoin-89e24fa6a670-arm-linux-gnueabihf.tar.gz
0170089a9b2b47791bcfb1430045e05ae68331c5b15147c59e859242af48998c  guix-build-89e24fa6a670/output/dist-archive/bitcoin-89e24fa6a670.tar.gz
f004cd8dda03e3aa3b5ad16902cf464d3a728800624bbb4805f10e46c76e083a  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/SHA256SUMS.part
39b2cb40782a7876f6d0ef6149a22b1cb1115359aeaf28b8747366ea056caeb1  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/bitcoin-89e24fa6a670-powerpc64-linux-gnu-debug.tar.gz
8b6811cf8f884cdac81cec3df3e7a8c8ec9602d3f0580972260ac13e511efbd0  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/bitcoin-89e24fa6a670-powerpc64-linux-gnu.tar.gz
be1b7128ac267e2c0a69d36e30fcaf1e5658280863d65a806db5bca5a1eedf26  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/SHA256SUMS.part
10e32cc185845f6bfc6754b3d9051d8e5887da76d610726cf575a6b38397ca96  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/bitcoin-89e24fa6a670-powerpc64le-linux-gnu-debug.tar.gz
34cfc0481e509fef84f413b3f1d99481f06d0059bad44e83bdfbf6492dd6161c  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/bitcoin-89e24fa6a670-powerpc64le-linux-gnu.tar.gz
a2524ca430d878589087e249832f4ca034b7d6a60f3aa3f360adbf068733faeb  guix-build-89e24fa6a670/output/riscv64-linux-gnu/SHA256SUMS.part
b355090dd832765237ed0d7dc3fe7929a52cf675624ec0ec04942fdd0f1deacc  guix-build-89e24fa6a670/output/riscv64-linux-gnu/bitcoin-89e24fa6a670-riscv64-linux-gnu-debug.tar.gz
39176b1a3f457f430e3b3f5c8a454b7f5a5da7dce8f83b19be1dd4ebb751992b  guix-build-89e24fa6a670/output/riscv64-linux-gnu/bitcoin-89e24fa6a670-riscv64-linux-gnu.tar.gz
d4c9a8886571f927c87dd4d57686d430fbc9ede90e407526277f16faf222aedb  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/SHA256SUMS.part
302ef30c220d8601f9bf141b37a38675ec27b91ff4a86e6da6ebc2617d1c17ef  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx-unsigned.dmg
1cb5e259a62a1d312e76ef078ef0be5d18355e864140072573a73c131bd6cb90  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx-unsigned.tar.gz
fa1c48d6d968c520a372672939285145c120c8802c009b9505552ea415086e37  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx64.tar.gz
8857b43bc6967aa4eb6c74922a63a5598cb7314931214f0aeb47faae1c03b39b  guix-build-89e24fa6a670/output/x86_64-linux-gnu/SHA256SUMS.part
f2de743e688a95f9aa3d90a3c4c3d42d47fefba79459d5cf0382608aff99c47a  guix-build-89e24fa6a670/output/x86_64-linux-gnu/bitcoin-89e24fa6a670-x86_64-linux-gnu-debug.tar.gz
15058102fc12b3bb263be3d4108e25e476e04806e96082c41689e8543b610114  guix-build-89e24fa6a670/output/x86_64-linux-gnu/bitcoin-89e24fa6a670-x86_64-linux-gnu.tar.gz
10c49d959ea5ee8fa102bba083cd36d7302cbbeeef2d05e861d8cc19c9898af0  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/SHA256SUMS.part
8ffd891ea61093bb7a7a7639da8dafca9c8543fd70a1d7edfb2d61b133cfa110  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win-unsigned.tar.gz
dbed9b02d5e8bb87cf39185795ede932c70b096a581b928aac4eacb9bd835e6c  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64-debug.zip
45d1f4e7ff026750f007489c7d9ee361b907f742d10ee95ed644ef8c8bdf696d  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64-setup-unsigned.exe
22f2a025afc2cc81c623c997e9c7d8fcf612555545547896992bb5de8c2ad6b7  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64.zip
```

sipa commented:
Yeah, it's getting close. We can wait.

hebasto commented:
#### Guix builds:

```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
cc137780b4627e896a7d4995c713938a94d7dad572c96e0b85fdd7ff368ca137  guix-build-89e24fa6a670/output/aarch64-linux-gnu/SHA256SUMS.part
48b964bd125987f2d2ad5718db564fc35fa0d8466464d78c5b3f3f7e7a0f4b6a  guix-build-89e24fa6a670/output/aarch64-linux-gnu/bitcoin-89e24fa6a670-aarch64-linux-gnu-debug.tar.gz
1ee6fe2021d4e0fb22048c63085f2701c678705def890671e20593adb030b882  guix-build-89e24fa6a670/output/aarch64-linux-gnu/bitcoin-89e24fa6a670-aarch64-linux-gnu.tar.gz
794d048626880d23ca8dafd0d199bc0ebb2d8d1bff6b6b43697c4cdc3c66ce4a  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/SHA256SUMS.part
90d91e3401f2d2182733d35570747fdb4499048f8c78fedf853f253031fbb83c  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/bitcoin-89e24fa6a670-arm-linux-gnueabihf-debug.tar.gz
eb2830ddd5303a6daa77f823dbc19ea7a8f0215aae1750d8b1c831adfa2e149a  guix-build-89e24fa6a670/output/arm-linux-gnueabihf/bitcoin-89e24fa6a670-arm-linux-gnueabihf.tar.gz
0170089a9b2b47791bcfb1430045e05ae68331c5b15147c59e859242af48998c  guix-build-89e24fa6a670/output/dist-archive/bitcoin-89e24fa6a670.tar.gz
f004cd8dda03e3aa3b5ad16902cf464d3a728800624bbb4805f10e46c76e083a  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/SHA256SUMS.part
39b2cb40782a7876f6d0ef6149a22b1cb1115359aeaf28b8747366ea056caeb1  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/bitcoin-89e24fa6a670-powerpc64-linux-gnu-debug.tar.gz
8b6811cf8f884cdac81cec3df3e7a8c8ec9602d3f0580972260ac13e511efbd0  guix-build-89e24fa6a670/output/powerpc64-linux-gnu/bitcoin-89e24fa6a670-powerpc64-linux-gnu.tar.gz
be1b7128ac267e2c0a69d36e30fcaf1e5658280863d65a806db5bca5a1eedf26  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/SHA256SUMS.part
10e32cc185845f6bfc6754b3d9051d8e5887da76d610726cf575a6b38397ca96  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/bitcoin-89e24fa6a670-powerpc64le-linux-gnu-debug.tar.gz
34cfc0481e509fef84f413b3f1d99481f06d0059bad44e83bdfbf6492dd6161c  guix-build-89e24fa6a670/output/powerpc64le-linux-gnu/bitcoin-89e24fa6a670-powerpc64le-linux-gnu.tar.gz
a2524ca430d878589087e249832f4ca034b7d6a60f3aa3f360adbf068733faeb  guix-build-89e24fa6a670/output/riscv64-linux-gnu/SHA256SUMS.part
b355090dd832765237ed0d7dc3fe7929a52cf675624ec0ec04942fdd0f1deacc  guix-build-89e24fa6a670/output/riscv64-linux-gnu/bitcoin-89e24fa6a670-riscv64-linux-gnu-debug.tar.gz
39176b1a3f457f430e3b3f5c8a454b7f5a5da7dce8f83b19be1dd4ebb751992b  guix-build-89e24fa6a670/output/riscv64-linux-gnu/bitcoin-89e24fa6a670-riscv64-linux-gnu.tar.gz
d4c9a8886571f927c87dd4d57686d430fbc9ede90e407526277f16faf222aedb  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/SHA256SUMS.part
302ef30c220d8601f9bf141b37a38675ec27b91ff4a86e6da6ebc2617d1c17ef  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx-unsigned.dmg
1cb5e259a62a1d312e76ef078ef0be5d18355e864140072573a73c131bd6cb90  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx-unsigned.tar.gz
fa1c48d6d968c520a372672939285145c120c8802c009b9505552ea415086e37  guix-build-89e24fa6a670/output/x86_64-apple-darwin19/bitcoin-89e24fa6a670-osx64.tar.gz
8857b43bc6967aa4eb6c74922a63a5598cb7314931214f0aeb47faae1c03b39b  guix-build-89e24fa6a670/output/x86_64-linux-gnu/SHA256SUMS.part
f2de743e688a95f9aa3d90a3c4c3d42d47fefba79459d5cf0382608aff99c47a  guix-build-89e24fa6a670/output/x86_64-linux-gnu/bitcoin-89e24fa6a670-x86_64-linux-gnu-debug.tar.gz
15058102fc12b3bb263be3d4108e25e476e04806e96082c41689e8543b610114  guix-build-89e24fa6a670/output/x86_64-linux-gnu/bitcoin-89e24fa6a670-x86_64-linux-gnu.tar.gz
10c49d959ea5ee8fa102bba083cd36d7302cbbeeef2d05e861d8cc19c9898af0  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/SHA256SUMS.part
8ffd891ea61093bb7a7a7639da8dafca9c8543fd70a1d7edfb2d61b133cfa110  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win-unsigned.tar.gz
dbed9b02d5e8bb87cf39185795ede932c70b096a581b928aac4eacb9bd835e6c  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64-debug.zip
45d1f4e7ff026750f007489c7d9ee361b907f742d10ee95ed644ef8c8bdf696d  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64-setup-unsigned.exe
22f2a025afc2cc81c623c997e9c7d8fcf612555545547896992bb5de8c2ad6b7  guix-build-89e24fa6a670/output/x86_64-w64-mingw32/bitcoin-89e24fa6a670-win64.zip
```

real-or-random commented:
https://github.com/bitcoin-core/secp256k1/pull/988 has been merged. @sipa Can you update the tree here?

sipa committed:
Squashed 'src/secp256k1/' changes from be8d9c262f..0559fc6e41

0559fc6e41 Merge bitcoin-core/secp256k1#988: Make signing table fully static
7dfceceea6 build: Remove #undef hack for ASM in the precomputation programs
bb36fe9be0 ci: Test `make precomp`
d94a37a20c build: Remove CC_FOR_BUILD stuff
ad63bb4c29 build: Prebuild and distribute ecmult_gen table
ac49361ed0 prealloc: Get rid of manual memory management for prealloc contexts
6573c08f65 ecmult_gen: Tidy precomputed file and save space
5eba83f17c ecmult_gen: Precompute tables for all values of ECMULT_GEN_PREC_BITS
5d0dbef018 Merge bitcoin-core/secp256k1#942: Verify that secp256k1_ge_set_gej_zinv does not operate on infinity.
486205aa68 Merge bitcoin-core/secp256k1#920: Test all ecmult functions with many j*2^i combinations
fdb33dd122 refactor: Make PREC_BITS a parameter of ecmult_gen_build_prec_table
5eb519e1f6 ci: reduce TEST_ITERS in memcheck run
e2cf77328a Test ecmult functions for all i*2^j for j=0..255 and odd i=1..255.
61ae37c612 Merge bitcoin-core/secp256k1#1022: build: Windows DLL additions
4f01840b82 Merge bitcoin-core/secp256k1#1027: build: Add a check that Valgrind actually supports a host platform
6ad908aa00 Merge bitcoin-core/secp256k1#1008: bench.c: add `--help` option and ci: move env variables
592661c22f ci: move test environment variable declaration to .cirrus.yml
dcbe84b841 bench: add --help option to bench.
099bad945e Comment and check a parameter for inf in secp256k1_ecmult_const.
6c0be857f8 Verify that secp256k1_ge_set_gej_zinv does not operate on infinity. a->x and a->y should not be used if the infinity flag is set.
4900227451 Merge bitcoin-core/secp256k1#1025: build: replace backtick command substitution with $()
7c7ce872a5 build: Add a check that Valgrind actually supports a host platform
a4875e30a6 refactor: Move default callbacks to util.h
4c94c55bce doc: Remove obsolete hint for valgrind stack size
5106226991 exhaustive_tests: Fix with ecmult_gen table with custom generator
e1a76530db refactor: Make generator a parameter of ecmult_gen_create_prec_table
9ad09f6911 refactor: Rename program that generates static ecmult_gen table
8ae18f1ab3 refactor: Rename file that contains static ecmult_gen table
00d2fa116e ecmult_gen: Make code consistent with comment
3b0c2185ea ecmult_gen: Simplify ecmult_gen context after making table static
2b7c7497ef build: replace backtick command substitution with $()
49f608de47 Merge bitcoin-core/secp256k1#1004: ecmult: fix definition of STRAUSS_SCRATCH_OBJECTS
c0cd7de6d4 build: add -no-undefined to libtool LDFLAGS
fe32a79d35 build: pass win32-dll to LT_INIT
60bf8890df ecmult: fix definition of STRAUSS_SCRATCH_OBJECTS
fecf436d53 Merge bitcoin-core/secp256k1#1019: build: don't append valgrind CPPFLAGS if not installed (macOS)
2e5e4b67df Merge bitcoin-core/secp256k1#1020: doc: remove use of <0xa0> "no break space"
812ff5c747 doc: remove use of 0xa0 "no break space"
214042a170 build: don't append valgrind CPPFLAGS if not installed
e43ba02cfc refactor: Decouple table generation and ecmult_gen context
22dc2c0a0d ecmult_gen: Move table creation to new file and force static prec
793ad9016a Merge bitcoin-core/secp256k1#1010: doc: Minor fixes in safegcd_implementation.md
dc9b6853b7 doc: Minor fixes in safegcd_implementation.md
ea5e8a9c47 Merge bitcoin-core/secp256k1#1012: Fix typos
233297579d Fix typos
7006f1b97f Merge bitcoin-core/secp256k1#1011: ci: Enable -g if we set CFLAGS manually
72de1359e9 ci: Enable -g if we set CFLAGS manually
74c34e727b Merge bitcoin-core/secp256k1#1009: refactor: Use (int)&(int) in boolean context to avoid compiler warning
16d132215c refactor: Use (int)&(int) in boolean context to avoid compiler warning
c74a7b7e51 Merge bitcoin-core/secp256k1#1007: doc: Replace apoelstra's GPG key by jonasnick's GPG key
3b157c48ed doc: Suggest keys.openpgp.org as keyserver in SECURITY.md
73a7472cd0 doc: Replace apoelstra's GPG key by jonasnick's GPG key
515a5dbd02 Merge bitcoin-core/secp256k1#991: Merge all "external" benchmarks into a single bench binary
af6abcb3d0 Make bench support selecting which benchmarks to run
9f56bdf5b9 Merge bench_schnorrsig into bench
3208557ae1 Merge bench_recover into bench
855e18d8a8 Merge bench_ecdh into bench
2a7be678a6 Combine bench_sign and bench_verify into single bench
8fa41201bd Merge bitcoin-core/secp256k1#1002: Make aux_rnd32==NULL behave identical to 0x0000..00.
5324f8942d Make aux_rnd32==NULL behave identical to 0x0000..00.
21c188b3c5 Merge bitcoin-core/secp256k1#943: VERIFY_CHECK precondition for secp256k1_fe_set_int.
3e7b2ea194 Merge bitcoin-core/secp256k1#999: bench_ecmult: improve clarity of output
23e2f66726 bench: don't return 1 in have_flag() if argc = 1
96b1ad2ea9 bench_ecmult: improve clarity of output
20d791edfb Merge bitcoin-core/secp256k1#989: Shared benchmark format for command line and CSV outputs
aa1b889b61 Merge bitcoin-core/secp256k1#996: Fix G.y parity in sage code
044d956305 Fix G.y parity in sage code
b4b130678d create csv file from the benchmark output
26a255beb6 Shared benchmark format for command line and CSV outputs
9526874d14 Merge bitcoin-core/secp256k1#810: Avoid overly-wide multiplications in 5x52 field mul/sqr
920a0e5fa6 Merge bitcoin-core/secp256k1#952: Avoid computing out-of-bounds pointer.
f34b5cae03 Merge bitcoin-core/secp256k1#983: [RFC] Remove OpenSSL testing support
297ce82091 Merge bitcoin-core/secp256k1#966: Make aux_rand32 arg to secp256k1_schnorrsig_sign const
2888640132 VERIFY_CHECK precondition for secp256k1_fe_set_int.
d49011f54c Make _set_fe_int( . , 0 ) set magnitude to 0
bc08599e77 Remove OpenSSL testing support
10f9bd84f4 Merge bitcoin-core/secp256k1#987: Fix unused parameter warnings when building without VERIFY
189f6bcfef Fix unused parameter warnings when building without VERIFY
da0092bccc Merge bitcoin-core/secp256k1#986: tests: remove `secp256k1_fe_verify` from tests.c and modify `_fe_from_storage` to call `_fe_verify`
d43993724d tests: remove `secp256k1_fe_verify` from tests.c and modify `secp256k1_fe_from_storage` to call `secp256k1_fe_verify`
2a3a97c665 Merge bitcoin-core/secp256k1#976: `secp256k1_schnorrsig_sign_internal` should be static
aa5d34a8fe Merge bitcoin-core/secp256k1#783: Make the public API docs more consistent and explicit
72713872a8 Add missing static to secp256k1_schnorrsig_sign_internal
db4667d5e0 Make aux_rand32 arg to secp256k1_schnorrsig_sign const
9a5a87e0f1 Merge bitcoin-core/secp256k1#956: Replace ecmult_context with a generated static array.
20abd52c2e Add tests for pre_g tables.
6815761cf5 Remove ecmult_context.
f20dcbbad1 Correct typo.
16a3cc07e8 Generate ecmult_static_pre_g.h
8de2d86a06 Bump memory limits in advance of making the ecmult context static.
d7ec49a689 Merge bitcoin-core/secp256k1#969: ci: Fixes after Debian release
5d5c74a057 tests: Rewrite code to circument potential bug in clang
3d2f492ceb ci: Install libasan6 (instead of 5) after Debian upgrade
adec5a1638 Add missing null check for ctx and input keys in the public API
f4edfc7581 Improve consistency for NULL arguments in the public interface
9be7b0f083 Avoid computing out-of-bounds pointer.
b53e0cd61f Avoid overly-wide multiplications

git-subtree-dir: src/secp256k1
git-subtree-split: 0559fc6e41b65af6e52c32eb9b1286494412a162

sipa committed:
Update secp256k1 subtree to latest upstream master

sipa commented:
Rebased, and updated to latest secp256k1 upstream master. Newly included changes:
* bitcoin-core/secp256k1#1020
* bitcoin-core/secp256k1#1019
* bitcoin-core/secp256k1#1004
* bitcoin-core/secp256k1#1025
* bitcoin-core/secp256k1#1008
* bitcoin-core/secp256k1#1027
* bitcoin-core/secp256k1#1022
* bitcoin-core/secp256k1#920
* bitcoin-core/secp256k1#942
* bitcoin-core/secp256k1#988

The last one is relevant, as it removes all compile-time precomputation and replaces it with precomputed tables that are included in the source code, simplifying the build system. The other changes are benchmark/testing/building related.

real-or-random commented:
> The last one is relevant, as it removes all compile-time precomputation and replaces it with precomputed tables that are included in the source code, simplifying the build system. The other changes are benchmark/testing/building related.

I just stumbled over this in https://github.com/bitcoin/bitcoin/blob/2560b6862b405c72daa03630c463673c3252020d/doc/build-windows.md?plain=1#L34-L35) and wonder if it can be removed now:
> A host toolchain (`build-essential`) is necessary because some dependency
> packages need to build host utilities that are used in the build process.

I suspect the answer is "no" because other also deps need this but I have no idea.

hebasto commented:
> I suspect the answer is "no" because other also deps need this but I have no idea.

E.g., `b2` build tool for `boost`, and `qmake` build tool for `qt`.

hebasto commented:
Ok, tested 314195c8be3bd7db0d5817c4fb3aa85c84363ce9.

#22854 resolved.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
1a271232a562cbf6c520e896e28fbe2db0fda2bb52e1a1c995ce9179f1814f60  guix-build-314195c8be3b/output/aarch64-linux-gnu/SHA256SUMS.part
148e5ccfde730dbf43c3638d5f66ffb94b1f5b708ae6dfffd047da92c0e2b8c2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu-debug.tar.gz
6f3a61f2f152862f117323e2f14cc5ece8c5fcd82f8960b6e028b82443e2e188  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu.tar.gz
34403666677362283a5a2a981e9b623a025cac618d6e555a5734e3eafcf2cdd1  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
47a2a900e0ee375c0039df67ee464e9b0d1a59016feea8f839b41f9f3cbe6949  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
ec762def21aaa3c122812b05b68a3c7ccbc6b8f199d35de4da8287b564cf43b8  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
bf31b9edb34b3afc8668fe52844a6bf8f993680d19eeb5fde9af4dfe357f6bbd  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
cf693015c70451e6182da9aa4892a62403b7fc0d670756118c3ca18fe3843944  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
8418cbad866f81448e4343205d2bdfed61244604eae3e809feee997e9b163b8a  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
fe9cb70cdbee18b56fb4a88b6b74020efbc20f88c8aa2433b30e5b202927148f  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
670825457acce2b5fd0de9cea2b784a78b2a4af3825d4b691d424a693dd26bfe  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
bb59bd423f7cd8bd1e4f047f5e1845d931266b8126aff6a8e0099595b8d5c5c3  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
1661cc8351ef6c36d4ab37e0ae3f0a3eacceb928d9a0953c7b45c31f53dcc0d4  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
0b6594a8feb20042c081dbc7868368d6c33258fd873cfc6d877085d7bb2f7932  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
e48490813395471b1f63628056b557e52b401be25a28ecc137208d47c142c0fa  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
e59251c31c806dcabcf272829ea7911a4cc2574f8a85d1882bb9297ab9798037  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
06664b118bac83ec1dbcf0c8c8c75e19c51e8e3a2d6f1a4ee19350cff0d37a9e  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
bb9a31fdd1a5f67213217539d4292c985f145481656d01b278f2beb9c7c41711  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ada7ae94c6172b8970a3de8927491b88f95ed1aa9262fbc0ebde604d30df595f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
f426af809642871784547ef2f2c2f58bc218309b6c77dc032d52ad92860ad80f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
abc7b36e5b2007ad1d5fda34af40f0be550b3f4d1f748ff2b8e118bdd2f7c69f  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
21ec8c6331c3c31e34941cf9703e5e06bf172be97e2817f04aad224e5d3bb153  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
cf5234cd7c5025fc786255729a40e02d613faf30e0994fc2b70094549f555fb1  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f90f5aeb4bde55276086ced0813d01bee663ee07af8d98a24fde1954f8ed099d  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
dd603e29b9454a55fbe25e793dc29e469dabf3d6e4c2cf28eb3986d46129614f  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

fanquake commented:
> Guix builds:

Seeing mismatches in the Guix build (except for darwin):
```bash
ba8d6ccf857a5145d70827992be2c68eb33ad447a7a1b29ae9f682a33ea3817a  guix-build-314195c8be3b/output/aarch64-linux-gnu/SHA256SUMS.part
148e5ccfde730dbf43c3638d5f66ffb94b1f5b708ae6dfffd047da92c0e2b8c2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu-debug.tar.gz
a76ea6d62e93f188dc324d45afd3622c37959efd8a40039b985310d9d0fdf7f2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu.tar.gz
2fe1c5e171437a213830acd17988493d1fe3e446857ea5c5dbc2d10a4196f617  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
47a2a900e0ee375c0039df67ee464e9b0d1a59016feea8f839b41f9f3cbe6949  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
dc22ae327c1d614ad991e8d63adb9d39c973de675a0087e4ee4e34feebcb0a8e  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
aabbe02de83be422c71b0885ff837fc7222eedc72c865e159c9b3240f3df0766  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
cf693015c70451e6182da9aa4892a62403b7fc0d670756118c3ca18fe3843944  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
ca6c992eb49c7cf3c638f81ca22c9e4c0cec486eff2dee0ee4369ebde33f51fc  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
a36a0c1a650e9bfd2dcaa6e26d24a62f9ff23ebcac0b629c8f8f156e28591f79  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
670825457acce2b5fd0de9cea2b784a78b2a4af3825d4b691d424a693dd26bfe  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
4f67258d0c6ad7d74b47e5ccbf58f7164b98d7767f0d0a9d61b84869f45cb7a5  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
60d6160cc98d9ce4b072cbc10f0ce307626abcb3b9c96a69ac2c066413097c2b  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
0b6594a8feb20042c081dbc7868368d6c33258fd873cfc6d877085d7bb2f7932  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
95a83a44b148cadee90bbd2747ef929d4a3ff98a295cf710b1f29e83ec0e5b14  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
e59251c31c806dcabcf272829ea7911a4cc2574f8a85d1882bb9297ab9798037  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
06664b118bac83ec1dbcf0c8c8c75e19c51e8e3a2d6f1a4ee19350cff0d37a9e  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
6139c52dd21f163f3a717aa98ddf597e2954592039096c02df342928bd610c40  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ada7ae94c6172b8970a3de8927491b88f95ed1aa9262fbc0ebde604d30df595f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
6fc3b34e834a77aea53322e274f41779225698a1094bd086c40e7cb1772abae0  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
d4719fcb6c51dfa405d803fa4b46e36ea62b02b4b4cac5da828028aad9e64f49  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
f75aea7524f732f13b08063e1d846ffc0cc5f232d982c6c340b54aad3fee2882  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
cf5234cd7c5025fc786255729a40e02d613faf30e0994fc2b70094549f555fb1  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f038729afec7c11283559cb50f2053bb125d75d234f6cfa4813789d28238d242  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
24de2ca9a64cfb1b6de92bd7cc3843357897dd4d4caca8dfe5fad1c3dde9f199  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

laanwj commented:
i built twice and got this each time (matches @fanquake, mismatches @hebasto's):
```
ba8d6ccf857a5145d70827992be2c68eb33ad447a7a1b29ae9f682a33ea3817a  guix-build-314195c8be3b/output/aarch64-linux-gnu/SHA256SUMS.part
148e5ccfde730dbf43c3638d5f66ffb94b1f5b708ae6dfffd047da92c0e2b8c2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu-debug.tar.gz
a76ea6d62e93f188dc324d45afd3622c37959efd8a40039b985310d9d0fdf7f2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu.tar.gz
2fe1c5e171437a213830acd17988493d1fe3e446857ea5c5dbc2d10a4196f617  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
47a2a900e0ee375c0039df67ee464e9b0d1a59016feea8f839b41f9f3cbe6949  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
dc22ae327c1d614ad991e8d63adb9d39c973de675a0087e4ee4e34feebcb0a8e  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
aabbe02de83be422c71b0885ff837fc7222eedc72c865e159c9b3240f3df0766  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
cf693015c70451e6182da9aa4892a62403b7fc0d670756118c3ca18fe3843944  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
ca6c992eb49c7cf3c638f81ca22c9e4c0cec486eff2dee0ee4369ebde33f51fc  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
a36a0c1a650e9bfd2dcaa6e26d24a62f9ff23ebcac0b629c8f8f156e28591f79  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
670825457acce2b5fd0de9cea2b784a78b2a4af3825d4b691d424a693dd26bfe  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
4f67258d0c6ad7d74b47e5ccbf58f7164b98d7767f0d0a9d61b84869f45cb7a5  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
60d6160cc98d9ce4b072cbc10f0ce307626abcb3b9c96a69ac2c066413097c2b  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
0b6594a8feb20042c081dbc7868368d6c33258fd873cfc6d877085d7bb2f7932  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
95a83a44b148cadee90bbd2747ef929d4a3ff98a295cf710b1f29e83ec0e5b14  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
e59251c31c806dcabcf272829ea7911a4cc2574f8a85d1882bb9297ab9798037  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
06664b118bac83ec1dbcf0c8c8c75e19c51e8e3a2d6f1a4ee19350cff0d37a9e  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
6139c52dd21f163f3a717aa98ddf597e2954592039096c02df342928bd610c40  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ada7ae94c6172b8970a3de8927491b88f95ed1aa9262fbc0ebde604d30df595f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
6fc3b34e834a77aea53322e274f41779225698a1094bd086c40e7cb1772abae0  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
d4719fcb6c51dfa405d803fa4b46e36ea62b02b4b4cac5da828028aad9e64f49  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
f75aea7524f732f13b08063e1d846ffc0cc5f232d982c6c340b54aad3fee2882  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
cf5234cd7c5025fc786255729a40e02d613faf30e0994fc2b70094549f555fb1  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f038729afec7c11283559cb50f2053bb125d75d234f6cfa4813789d28238d242  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
24de2ca9a64cfb1b6de92bd7cc3843357897dd4d4caca8dfe5fad1c3dde9f199  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

hebasto commented:
After `make -C depends clean-all` I've got:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
ba8d6ccf857a5145d70827992be2c68eb33ad447a7a1b29ae9f682a33ea3817a  guix-build-314195c8be3b/output/aarch64-linux-gnu/SHA256SUMS.part
148e5ccfde730dbf43c3638d5f66ffb94b1f5b708ae6dfffd047da92c0e2b8c2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu-debug.tar.gz
a76ea6d62e93f188dc324d45afd3622c37959efd8a40039b985310d9d0fdf7f2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu.tar.gz
2fe1c5e171437a213830acd17988493d1fe3e446857ea5c5dbc2d10a4196f617  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
47a2a900e0ee375c0039df67ee464e9b0d1a59016feea8f839b41f9f3cbe6949  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
dc22ae327c1d614ad991e8d63adb9d39c973de675a0087e4ee4e34feebcb0a8e  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
aabbe02de83be422c71b0885ff837fc7222eedc72c865e159c9b3240f3df0766  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
cf693015c70451e6182da9aa4892a62403b7fc0d670756118c3ca18fe3843944  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
ca6c992eb49c7cf3c638f81ca22c9e4c0cec486eff2dee0ee4369ebde33f51fc  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
a36a0c1a650e9bfd2dcaa6e26d24a62f9ff23ebcac0b629c8f8f156e28591f79  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
670825457acce2b5fd0de9cea2b784a78b2a4af3825d4b691d424a693dd26bfe  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
4f67258d0c6ad7d74b47e5ccbf58f7164b98d7767f0d0a9d61b84869f45cb7a5  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
60d6160cc98d9ce4b072cbc10f0ce307626abcb3b9c96a69ac2c066413097c2b  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
0b6594a8feb20042c081dbc7868368d6c33258fd873cfc6d877085d7bb2f7932  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
95a83a44b148cadee90bbd2747ef929d4a3ff98a295cf710b1f29e83ec0e5b14  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
e59251c31c806dcabcf272829ea7911a4cc2574f8a85d1882bb9297ab9798037  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
06664b118bac83ec1dbcf0c8c8c75e19c51e8e3a2d6f1a4ee19350cff0d37a9e  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
6139c52dd21f163f3a717aa98ddf597e2954592039096c02df342928bd610c40  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ada7ae94c6172b8970a3de8927491b88f95ed1aa9262fbc0ebde604d30df595f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
6fc3b34e834a77aea53322e274f41779225698a1094bd086c40e7cb1772abae0  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
d4719fcb6c51dfa405d803fa4b46e36ea62b02b4b4cac5da828028aad9e64f49  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
f75aea7524f732f13b08063e1d846ffc0cc5f232d982c6c340b54aad3fee2882  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
cf5234cd7c5025fc786255729a40e02d613faf30e0994fc2b70094549f555fb1  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f038729afec7c11283559cb50f2053bb125d75d234f6cfa4813789d28238d242  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
24de2ca9a64cfb1b6de92bd7cc3843357897dd4d4caca8dfe5fad1c3dde9f199  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

that matches to @fanquake's and @laanwj's results.

jarolrod commented:
GUIX hashes, mine match @hebasto's last hashes, @laanwj, and @fanquake hashes:

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

ba8d6ccf857a5145d70827992be2c68eb33ad447a7a1b29ae9f682a33ea3817a  guix-build-314195c8be3b/output/aarch64-linux-gnu/SHA256SUMS.part
148e5ccfde730dbf43c3638d5f66ffb94b1f5b708ae6dfffd047da92c0e2b8c2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu-debug.tar.gz
a76ea6d62e93f188dc324d45afd3622c37959efd8a40039b985310d9d0fdf7f2  guix-build-314195c8be3b/output/aarch64-linux-gnu/bitcoin-314195c8be3b-aarch64-linux-gnu.tar.gz
2fe1c5e171437a213830acd17988493d1fe3e446857ea5c5dbc2d10a4196f617  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
47a2a900e0ee375c0039df67ee464e9b0d1a59016feea8f839b41f9f3cbe6949  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
dc22ae327c1d614ad991e8d63adb9d39c973de675a0087e4ee4e34feebcb0a8e  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
aabbe02de83be422c71b0885ff837fc7222eedc72c865e159c9b3240f3df0766  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
cf693015c70451e6182da9aa4892a62403b7fc0d670756118c3ca18fe3843944  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
ca6c992eb49c7cf3c638f81ca22c9e4c0cec486eff2dee0ee4369ebde33f51fc  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
a36a0c1a650e9bfd2dcaa6e26d24a62f9ff23ebcac0b629c8f8f156e28591f79  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
670825457acce2b5fd0de9cea2b784a78b2a4af3825d4b691d424a693dd26bfe  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
4f67258d0c6ad7d74b47e5ccbf58f7164b98d7767f0d0a9d61b84869f45cb7a5  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
60d6160cc98d9ce4b072cbc10f0ce307626abcb3b9c96a69ac2c066413097c2b  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
0b6594a8feb20042c081dbc7868368d6c33258fd873cfc6d877085d7bb2f7932  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
95a83a44b148cadee90bbd2747ef929d4a3ff98a295cf710b1f29e83ec0e5b14  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
e59251c31c806dcabcf272829ea7911a4cc2574f8a85d1882bb9297ab9798037  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
06664b118bac83ec1dbcf0c8c8c75e19c51e8e3a2d6f1a4ee19350cff0d37a9e  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
6139c52dd21f163f3a717aa98ddf597e2954592039096c02df342928bd610c40  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ada7ae94c6172b8970a3de8927491b88f95ed1aa9262fbc0ebde604d30df595f  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
6fc3b34e834a77aea53322e274f41779225698a1094bd086c40e7cb1772abae0  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
d4719fcb6c51dfa405d803fa4b46e36ea62b02b4b4cac5da828028aad9e64f49  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
f75aea7524f732f13b08063e1d846ffc0cc5f232d982c6c340b54aad3fee2882  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
cf5234cd7c5025fc786255729a40e02d613faf30e0994fc2b70094549f555fb1  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f038729afec7c11283559cb50f2053bb125d75d234f6cfa4813789d28238d242  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
24de2ca9a64cfb1b6de92bd7cc3843357897dd4d4caca8dfe5fad1c3dde9f199  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

fanquake reviewed (approved):
ACK 314195c8be3bd7db0d5817c4fb3aa85c84363ce9 - this includes a nice simplification to the lilbsecp build system (and thus our build system), and fixes issues like #22854. Did a Guix build on x86 (above), as well as a build on arm64 (except for the arm64 host):
```bash
1e32a19bbfcc45177e11dfcc6186b4d90eda76d754431769b2f4b88db55dc8d9  guix-build-314195c8be3b/output/arm-linux-gnueabihf/SHA256SUMS.part
b6930cd955fb0ef5e6dd8ee7bf272694d67e39665b8eb3024f7e21e0268cc329  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf-debug.tar.gz
e83a4e59f8ee4d60a33683e16c67dab572e358d587fb9bef93548b082693f65e  guix-build-314195c8be3b/output/arm-linux-gnueabihf/bitcoin-314195c8be3b-arm-linux-gnueabihf.tar.gz
99b0228fb410a8388717391c96d44b94a6791f1c9c6cce82bf86d0e40c891417  guix-build-314195c8be3b/output/dist-archive/bitcoin-314195c8be3b.tar.gz
34894aa6da4b7756ce82946a6359772fa003f6cae2cab9d73dfe51bc52283e1d  guix-build-314195c8be3b/output/powerpc64-linux-gnu/SHA256SUMS.part
10a19dd0a75d537c04354776020b96da294e3a1812418ee334d88bfca03f9cd1  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu-debug.tar.gz
01b401683febf875ddd686029d8c99076ae8521e55aca8cb595321da86ee0380  guix-build-314195c8be3b/output/powerpc64-linux-gnu/bitcoin-314195c8be3b-powerpc64-linux-gnu.tar.gz
2764a3c6f945753b246bdcdcd2f72524490b41ae60060e3bb42d4adeeafb74c3  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/SHA256SUMS.part
97e643980c9f7a53f5456367bb3ca7e9919a6ee8e51edbd44edfa6a69d536fe4  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu-debug.tar.gz
82257d9f3f3ee92cb87921bbc894598beee35ec854b91a52914597a8fc737bd0  guix-build-314195c8be3b/output/powerpc64le-linux-gnu/bitcoin-314195c8be3b-powerpc64le-linux-gnu.tar.gz
9f2e6d4c1fe828f4806af57060cebbdb13d5b888606992c126194cb894894d2d  guix-build-314195c8be3b/output/riscv64-linux-gnu/SHA256SUMS.part
8c4186b77d77669a3363111c7df4afc2094fb9d18ffec58379e657c777a87fc0  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu-debug.tar.gz
8d30867c6e91763d7a5532bb11e90ceffc9522b9094d5d28427648b1d12310b9  guix-build-314195c8be3b/output/riscv64-linux-gnu/bitcoin-314195c8be3b-riscv64-linux-gnu.tar.gz
1925e4f0f51d818e853d300805fd62c2f4daafe16ae3cc999e785300fe48eda7  guix-build-314195c8be3b/output/x86_64-apple-darwin/SHA256SUMS.part
ba7b7b2c31b55899db00aa2128027cd67556d8cf4353ee343456e88a65f29ff2  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.dmg
c2ea7cde40a2d5fa5f3cc923db8b47c22a6b2f205ab8e93c9779928cd0dee4a1  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx-unsigned.tar.gz
e8db65a348c1c59e2ff58a817c1d38c4f173c4798ee0db7500584af71093c731  guix-build-314195c8be3b/output/x86_64-apple-darwin/bitcoin-314195c8be3b-osx64.tar.gz
683766ebcb3f0bcdfcf5099a546cbbd67772471c0c1c400064cfb9279cf6d598  guix-build-314195c8be3b/output/x86_64-linux-gnu/SHA256SUMS.part
ee9dd8c23e468075b6ca6cc1b83833ab55efd323ba1d004e927337e27f30d656  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu-debug.tar.gz
9d39fa6e9fc07e469a3b2623c003d098ba38d657acfdea0f801f30d5639a4343  guix-build-314195c8be3b/output/x86_64-linux-gnu/bitcoin-314195c8be3b-x86_64-linux-gnu.tar.gz
40846101e0b43cc173d01c2cd5a638f364c1874b7f525e586d403b8833908261  guix-build-314195c8be3b/output/x86_64-w64-mingw32/SHA256SUMS.part
f75aea7524f732f13b08063e1d846ffc0cc5f232d982c6c340b54aad3fee2882  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win-unsigned.tar.gz
33461d03224f70945c0290ec93aed5fa522b459c49375573f35f79869788ca85  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-debug.zip
f038729afec7c11283559cb50f2053bb125d75d234f6cfa4813789d28238d242  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64-setup-unsigned.exe
b050e5b5afc97ce5a95445ed8b79f9724c8e034189e7c208f4081750c28cab6b  guix-build-314195c8be3b/output/x86_64-w64-mingw32/bitcoin-314195c8be3b-win64.zip
```

Checked the subtree
```bash
src/secp256k1 in HEAD currently refers to tree f7211a90cdfcdce0259552b839cc5ed816ff9bb7
src/secp256k1 in HEAD was last updated in commit 86dbc4d075decb82fbba837aaa283cf0561897ad (tree f7211a90cdfcdce0259552b839cc5ed816ff9bb7)
GOOD
```
